torch/distributed/_tensor/ops/math_ops.py

```diff
@@ -518,3 +541,15 @@ def _replicate_dims_start_at(
         else:
             new_placements.append(p)  # keep the placement
     return tuple(new_placements)
+
+
+def _replicate_dim_at(
+    placements: Sequence[Placement], dim: int = 0
+) -> Tuple[Placement, ...]:
+    new_placements: List[Placement] = []
+    for p in placements:
+        if p.is_partial() or (isinstance(p, Shard) and p.dim == dim):
+            new_placements.append(Replicate())  # make it replicate
+        else:
+            new_placements.append(p)  # keep the placement
+    return tuple(new_placements)
```

Comment:
Maybe we can reuse the `replicate_reduction_dims` function in `math_ops.py` by calling `replicate_reduction_dims(placements, [dim])`. But this can be a BE work in another PR:
move `replicate_reduction_dims` to `_tensor/ops/utils.py`. Keep or replace `_replicate_dims_start_at` and `_replicate_dim_at` with that util function.